jbmc/src/java_bytecode/java_bytecode_convert_class.cpp

```diff
@@ -886,6 +908,39 @@ void add_java_array_types(symbol_tablet &symbol_table)
     java_new_array.copy_to_operands(old_length);
     code_assignt create_blank(local_symexpr, java_new_array);
 
+    codet copy_type_information = code_skipt();
+    if(l == 'a')
+    {
+      // Reference arrays carry additional type information in their classids
+      // which should be copied:
+      const auto &array_dimension_component =
+        class_type.get_component(JAVA_ARRAY_DIMENSION_FIELD_NAME);
+      const auto &array_element_classid_component =
+        class_type.get_component(JAVA_ARRAY_ELEMENT_CLASSID_FIELD_NAME);
+
+      member_exprt old_array_dimension(
+        old_array,
+        array_dimension_component.get_name(),
```

Comment:
:bulb: Sorry should have been more explicit - this is definitely better, but you can even go one further:
```suggestion
        member_exprt old_array_dimension(old_array, array_dimension_component)
```
using the constructor that takes a `struct_typet::componentt` directly